src/Sylius/Bundle/ApiBundle/Serializer/PromotionDenormalizer.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Sylius\Bundle\ApiBundle\Serializer;
+
+use Sylius\Bundle\ApiBundle\Helpers\PromotionDataPricesTransformer;
+use Sylius\Component\Core\Model\Promotion;
+use Symfony\Component\Serializer\Normalizer\ContextAwareDenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+class PromotionDenormalizer implements ContextAwareDenormalizerInterface
+{
+    /**
+     * @var DenormalizerInterface
+     */
+    private DenormalizerInterface $objectNormalizer;
+    private string $classType;
+
+    public function __construct(
+        DenormalizerInterface $objectNormalizer,
+        string $classType
+    )
+    {
+        $this->objectNormalizer = $objectNormalizer;
+        $this->classType = $classType;
+    }
+
+    public function denormalize($data, string $type, string $format = null, array $context = [])
+    {
+        $context['api_allow_update'] = true;
+
+        $transformer = new PromotionDataPricesTransformer(false);
```

Comment:
Try to avoid create a new instance, because api platform use this class as service, just inject it whenever it need for you

Reply:
I need this class only here